flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -477,8 +477,8 @@ public boolean isFinished() {
 
 	@Override
 	public void requestPartitions() throws IOException, InterruptedException {
-		synchronized (requestLock) {
-			if (!requestedPartitionsFlag) {
+		if (!requestedPartitionsFlag) {
```

Comment:
1. please revert order of if/else for simplicity
2. you are effectively making this method not thread safe, which might be some "gotcha" moment for some future developer

Reply:
1. yes, there is this general rule that having shorter/simpler branch first:
```
if (condition) {
  one_liner
}
else {
 sth
 very
 long
 that 
 do
 not
 fit
 in
 programmer
 brain's
 cache
}
```
 is better, since once reader gets to the else, he can already forget about it and focus on the else branch, since there is "all of the remaining logic". Putting it into extreme (with `zero_liner` else branch), your example:
```
if (condition) {
 sth
 very
 long
 that 
 while
 reading
 one
 is
 all
 the
 time
 unsure
 what
 will
 happen
 in
 else
 branch
}
```

2. you are right, it doesn't matter.

% if/else branches order, LGTM

Reply:
ok, done (and there goes the 2-line patch :( )